client/src/actions/index.ts

```diff
@@ -258,6 +288,84 @@ const requestDifferentialExpression =
     }
   };
 
+/*
+ Check local storage for flag indicating that the work in progress toast should be displayed.
+ */
+export const checkExplainNewTab =
+  () =>
+  (dispatch: AppDispatch): void => {
+    const workInProgressWarn = storageGetTransient(KEYS.WORK_IN_PROGRESS_WARN);
+    if (workInProgressWarn) {
+      dispatch({ type: "work in progress warning displayed" });
+      postExplainNewTab(
+        "To maintain your in-progress work on the previous dataset, we opened this dataset in a new tab."
+      );
+    }
+  };
+
+/*
+ Handle select of dataset from dataset selector: determine whether to display dataset in current browser tab or open 
+ dataset in new tab if user currently has work in progress.
+ @param dataset Dataset to switch to and load in the current tab.
+ */
+export const selectDataset =
+  (dataset: Dataset): ((dispatch: AppDispatch, getState: GetState) => void) =>
+  (dispatch: AppDispatch, getState: GetState) => {
+    const workInProgress = selectIsUserStateDirty(getState());
+    if (workInProgress) {
+      dispatch(openDataset(dataset));
+    } else {
+      dispatch(switchDataset(dataset));
+    }
+  };
+
+/*
+ Open selected dataset in a new tab. Create local storage with expiry to pop toast once dataset is opened.
+ @param dataset Dataset to open in new tab.
+ */
+const openDataset =
+  (dataset: Dataset): ((dispatch: AppDispatch) => void) =>
+  (dispatch: AppDispatch) => {
+    const deploymentUrl = dataset.dataset_deployments?.[0].url ?? "";
```

Comment:
I believe the ` ?? ""` isn't needed, if the left side is undefined the if at the next line will trigger.

Reply:
[Updated to remove `??`](https://github.com/chanzuckerberg/single-cell-explorer/pull/111/files#diff-b00e58c03e2140bc121d75a4d38a78a3f3a7b83f31287a491e8c3e970ab4a7b4R329), thanks Emanuele!